views/default/lightbox.js.php

```diff
@@ -30,88 +23,89 @@
-	}
+				lightbox.bind(".elgg-lightbox");
+				lightbox.bind(".elgg-lightbox-photo", {photo: true});
+				lightbox.bind(".elgg-lightbox-inline", {inline: true});
+				lightbox.bind(".elgg-lightbox-iframe", {iframe: true});
 
-	elgg.ui.lightbox.bind(".elgg-lightbox");
-	elgg.ui.lightbox.bind(".elgg-lightbox-photo", {photo: true});
+				function registerDeprecationError() {
+					elgg.register_error("fancybox lightbox has been replaced by colorbox", 9999999999999);
+				}
 
-	if (typeof $.fancybox === 'undefined') {
-		$.fancybox = {
-			// error message for firefox users
-			__noSuchMethod__ : registerDeprecationError,
-			close: function () {
-				registerDeprecationError();
-				$.colorbox.close();
-			}
-		};
-		// support $().fancybox({type:'image'})
-		$.fn.fancybox = function (arg) {
-			registerDeprecationError();
-			if (arg.type === 'image') {
-				arg.photo = true;
-			}
-			this.colorbox(arg);
-			return this;
+				if (typeof $.fancybox === 'undefined') {
+					$.fancybox = {
+						// error message for firefox users
+						__noSuchMethod__: registerDeprecationError,
+						close: function () {
+							registerDeprecationError();
+							$.colorbox.close();
+						}
+					};
+					// support $().fancybox({type:'image'})
+					$.fn.fancybox = function (arg) {
+						registerDeprecationError();
+						if (arg.type === 'image') {
+							arg.photo = true;
+						}
+						this.colorbox(arg);
+						return this;
+					};
+				}
+			});
 		};
-	}
-};
-
-/**
- * Bind colorbox lightbox click to HTML
- *
- * @param {Object} selector CSS selector matching colorbox openers
- * @param {Object} opts     Colorbox options. These are overridden by data-colorbox-opts options
- */
-elgg.ui.lightbox.bind = function (selector, opts) {
-	if (!$.isPlainObject(opts)) {
-		opts = {};
-	}
-
-	// merge opts into defaults
-	opts = $.extend({}, elgg.ui.lightbox.getSettings(), opts);
 
-	$(document).on('click', selector, function (e) {
-		var $this = $(this),
-			href = $this.prop('href') || $this.prop('src'),
-			dataOpts = $this.data('colorboxOpts');
-		// Note: data-colorbox was reserved https://github.com/jackmoore/colorbox/issues/435
+		elgg.provide('elgg.ui.lightbox');
 
-		if (!$.isPlainObject(dataOpts)) {
-			dataOpts = {};
+		if (typeof elgg.ui.lightbox.getSettings === 'function') {
+			elgg.ui.lightbox.deprecated_settings = elgg.ui.lightbox.getSettings();
 		}
 
-		if (!dataOpts.href && href) {
-			dataOpts.href = href;
-		}
-
-		// merge data- options into opts
-		$.colorbox($.extend({}, opts, dataOpts));
-		e.preventDefault();
-	});
-};
-
-/**
- * Close the colorbox
- *
- */
-elgg.ui.lightbox.close = function() {
-	$.colorbox.close();
-};
+		/**
+		 * Lightbox settings
+		 * @deprecated 2.2
+		 */
+		elgg.ui.lightbox.getSettings = function () {
+			elgg.deprecated_notice('elgg.ui.lightbox.getSettings() has been deprecated and should not be called directly. Use elgg/lightbox AMD module intsead', '2.2');
+			var lightbox = require('elgg/lightbox');
```

Comment:
It did, but I suppose because we load elgg/lightbox in init
On Apr 5, 2016 4:26 PM, "Steve Clay" <notifications@github.com> wrote:

> In views/default/lightbox.js.php
> <https://github.com/Elgg/Elgg/pull/9611#discussion_r58547112>:
>
> > -};
> > -
> > -/**
> > - * Close the colorbox
> > - *
> > - */
> > -elgg.ui.lightbox.close = function() {
> > -	$.colorbox.close();
> > -};
> > +		/**
> > +		 * Lightbox settings
> > +		 * @deprecated 2.2
> > +		 */
> > +		elgg.ui.lightbox.getSettings = function () {
> > +			elgg.deprecated_notice('elgg.ui.lightbox.getSettings() has been deprecated and should not be called directly. Use elgg/lightbox AMD module intsead', '2.2');
> > +			var lightbox = require('elgg/lightbox');
>
> Have you tested that sync require('module') actually works?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9611/files/c3a290dc30d2aa697777932e6f0750bb5615044d#r58547112>
>